sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java

```diff
@@ -0,0 +1,503 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.models.UserDelegationKey;
+import com.azure.storage.common.Constants;
+import com.azure.storage.common.IPRange;
+import com.azure.storage.common.SASProtocol;
+import com.azure.storage.common.Utility;
+import com.azure.storage.common.credentials.SharedKeyCredential;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.OffsetDateTime;
+
+/**
+ * This class provides a fluent API to help aid in the creation of Azure Blob SAS Tokens.
+ */
+public class BlobSasConstructor {
```

Comment:
'Constructor' feels like an unfortunate suffix.

Reply:
I almost feel compelled to let you call this a Builder with the required CheckStyle suppression.

Reply:
I think the `Builder` suffix feels more fluid. Will get additional input on this class before continuing forward on implementing this and using it in Files and Queues.